core/indexer/common.go

```diff
@@ -98,6 +100,8 @@ func (cm *commonProcessor) buildTransaction(
 	header data.HeaderHandler,
 	txStatus string,
 ) *Transaction {
+	gasUsed := cm.minGasLimit + uint64(len(tx.Data))*cm.gasPerDataByte
```

Comment:
For invalid transactions, after the Mainnet upgrade, this will not be correct anymore. Perhaps use the soft fork swtich here as well (`PenalizedTooMuchGasEnableEpoch` - @sasurobert, @SebastianMarian is this the one?).

After the Mainnet upgrade, for invalid transactions, `gasUsed := gasLimit`.

Reply:
for invalid transactions gasUsed will be deduced from the receipt.